internal/backend/telemetry/meter.go

```diff
@@ -0,0 +1,83 @@
+package telemetry
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/embedded"
+)
+
+type namedMeter struct {
+	embedded.Meter
+
+	underlying metric.Meter
+	svcName    string
+}
+
+func (m *namedMeter) canonizeName(name string) string {
+	if !strings.HasPrefix(name, m.svcName) {
+		name = fmt.Sprintf("%s.%s", name, name)
+	}
+	return name
+}
+
+func (m *namedMeter) Int64Counter(name string, options ...metric.Int64CounterOption) (metric.Int64Counter, error) {
```

Comment:
Do we really use all of these counter types?

Reply:
these are all defined in the interface, so i had to delegate all.